scripts/courseGenerator/FieldworkCourseTwoSided.lua

```diff
@@ -26,6 +26,11 @@ function FieldworkCourseTwoSided:init(context)
     self.logger = Logger('FieldworkCourseTwoSided')
     context:setRowPattern(CourseGenerator.RowPatternAlternatingFirstRowEntryOnly())
     self:_setContext(context)
+    -- Add nil check for self.boundary
+    if not self.boundary then
+        self.context:addError(self.logger, 'Boundary is nil')
+        return
```

Comment:
How exactly this fixes the problem? It may prevent a crash, but definitely won't generate a course.